### DISCUSSION TITLE
How to remove this extra information in mini.completion for java

### DISCUSSION THREAD

**Discussion by 231tr0n** [nvim-mini/mini.nvim#1315] [Q&A] [answered]
Hi echasnovski! I came across an issue with completion items listed in mini.completion and specifically java.
Here is a screenshot of it
![image](https://github.com/user-attachments/assets/ed923ed7-44d5-4554-ab0c-43ccbb1931d7)
The thing is each completion item is very long for java setup using nvim-jdtls plugin which inturn makes the doc window not so good to read.
![image](https://github.com/user-attachments/assets/b4e67c9f-8f07-4950-b834-0258d193eda7)
So I wanted to know if there is any way in mini.completion to remove the extra part of completion items after completion kind and probably add it in the doc window as a heading if possible in any way.
Is something like this possible with mini.completion. I am currently struggling with this issue at work since the completion items itself cover most of the screen and the doc window is literally unreadable.

Do you think you there is any workaround for this issue?

**Comment by echasnovski** [comment, marked as answer]
That part seems to be a `detail` field of what server returns and it is used as `menu` part of completion item.

I think this will be resolved after #1258. In particular, see [this comment](https://github.com/echasnovski/mini.nvim/issues/1258#issuecomment-2395393286).

In the meantime, you can use [`config.lsp_completion.process_items`](https://github.com/echasnovski/mini.nvim/blob/0a8a1072137d916406507c941698a4bfa9dbbe7a/doc/mini-completion.txt#L218) to modify items received from LSP server. I think something like this should be enough:

```lua
require('mini.completion').setup({
  lsp_completion = {
    process_items = function(items, base)
      for _, i in ipairs(items) do
        i.detail = nil
      end
      return MiniCompletion.default_process_items(items, base)
    end
  }
})
```

As an alternative, you can also set `vim.b.minicompletion_config = { lsp_completion = { process_items = ... }  }` locally only for `java` buffers.

**Reply by 231tr0n** [reply]
Thank you soo much for helping me out. This works like a charm.
